sycl/test-e2e/syclcompat/group_utils/exchange.cpp

```diff
@@ -138,10 +139,11 @@ bool test_striped_to_blocked() {
       std::ostream_iterator<int> Iter(std::cout, ", ");
       std::copy(d_data, d_data + 512, Iter);
       std::cout << std::endl;
+      syclcompat::wait_and_free(d_data, q_ct1);
```

Comment:
nit: in these cases you've actually already waited for all the queues above so you can safely free the `d_data` etc

Reply:
nit: above in `dev_ct1.queues_wait_and_throw();` you could just wait on the queue you have used instead of waiting for all device queues.

Reply:
Ok, fixed.